Automattic/newspack-newsletters
# fix: sync newsletter campaigns on post meta update for all ESPs

### All Submissions:






### Changes proposed in this Pull Request:



#1526 changed the hook used to sync newsletter posts to the connected ESP to `updated_post_meta`. However, this was erroneously done only for Mailchimp and not the other ESPs.

### How to test the changes in this Pull Request:

1. On `release`, switch your ESP to ~a non-Mailchimp ESP~ Constant Contact.
2. Create a new newsletter and observe a message that never resolves in the sidebar: 

![cleanshot_2024-07-03_at_10 51 57_2x](https://github.com/Automattic/newspack-newsletters/assets/2230142/60b34239-985a-43d1-8811-aa024a39bc06)

3. Also observe that in the connected Constant Contact account, a synced campaign is never created.
4. Switch your ESP to Active Campaign.
5. Create a new newsletter and edit the content, then save. Observe that in the synced campaign in Active Campaign, the newsletter content is not updated after the save.
6. Check out this branch, switch back to CC, create another new newsletter, and confirm that the campaign syncs correctly for new newsletters. (You can also refresh and save the original newsletter created in step 1 and confirm that it gets synced upon save).
7. Switch to AC, then edit and save the newsletter you created in step 5. Confirm that the synced campaign in AC is updated after saving.

### Other information:
